pkg/collector/metric/container_metric.go

```diff
@@ -274,6 +274,9 @@ func (c *ContainerMetrics) UpdateCgroupMetrics() error {
 }
 
 func (c *ContainerMetrics) GetDynEnergyStat(component string) (energyStat *types.UInt64Stat) {
+	if c == nil {
+		return
+	}
```

Comment:
This shouldn't be done we must expect the receiver to be initialised at all time. If it is not the case, then it is a programming error and must panic. By returning we are only shadowing a logical error.